gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveCopyEntityHelper.java

```diff
@@ -609,6 +612,41 @@ private int addDeregisterSteps(List<CopyEntity> copyEntities, String fileSet, in
     copyEntities.add(new PrePublishStep(fileSet, Maps.<String, Object>newHashMap(), deregister, stepPriority++));
     return stepPriority;
   }
+  
+  private int addTableDeregisterSteps(List<CopyEntity> copyEntities, String fileSet, int initialPriority,
+      Table table) throws IOException {
+
+    int stepPriority = initialPriority;
+
+    Collection<Path> tablePaths = Lists.newArrayList();
+    DeregisterFileDeleteMethod deleteMethod = this.dataset.properties.containsKey(DELETE_FILES_ON_DEREGISTER)
+        ? DeregisterFileDeleteMethod.valueOf(this.dataset.properties.getProperty(DELETE_FILES_ON_DEREGISTER).toUpperCase())
+        : DeregisterFileDeleteMethod.NO_DELETE;
+
+    if (deleteMethod == DeregisterFileDeleteMethod.RECURSIVE) {
```

Comment:
I suggest using "switch" instead of "if" statement. As a safeguard, have a default clause to catch cases where new constants have been added but they are unprocessed.